cmd/console-proxy/src/posix.rs

```diff
@@ -0,0 +1,401 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+mod picocom_map;
+
+use std::fs::File;
+use std::io::{Read, Write};
+use std::os::unix::io::AsRawFd;
+use std::time::Duration;
+use std::{io, thread};
+
+use anyhow::{anyhow, bail, Context, Result};
+use clap::Parser;
+use crossbeam_channel::{select, Receiver, Sender};
+use picocom_map::RemapRules;
+use termios::Termios;
+
+use cmd_hiffy as humility_cmd_hiffy;
+
+use cmd_hiffy::HiffyLease;
+use humility::cli::Subcommand;
+use humility::core::Core;
+use humility::hubris::HubrisArchive;
+use humility_cmd::hiffy::HiffyContext;
+use humility_cmd::idol::{IdolArgument, IdolOperation};
+
+use super::UartConsoleArgs;
+use super::UartConsoleCommand;
+
+const HIFFY_BUF_SIZE: usize = 256;
+
+struct UartConsoleHandler<'a> {
+    hubris: &'a HubrisArchive,
+    core: &'a mut dyn Core,
+    context: HiffyContext<'a>,
+    poll_interval: Duration,
+}
+
+impl<'a> UartConsoleHandler<'a> {
+    pub fn new(
+        hubris: &'a HubrisArchive,
+        core: &'a mut dyn Core,
+        hiffy_timeout: u32,
+        poll_interval: u32,
+    ) -> Result<Self> {
+        let context = HiffyContext::new(hubris, core, hiffy_timeout)?;
+        Ok(Self {
+            hubris,
+            core,
+            context,
+            poll_interval: Duration::from_millis(u64::from(poll_interval)),
+        })
+    }
+
+    fn read(&mut self, buf: &mut [u8]) -> Result<usize> {
```

Comment:
Nit: rename to `uart_read` to make it obvious in `attach`

(same with `fn write` below)